src/main/java/org/opensearch/neuralsearch/processor/chunker/ChunkerUtil.java

```diff
@@ -18,24 +14,14 @@ public class ChunkerUtil {
     private ChunkerUtil() {} // no instance of this util class
 
     /**
-     * Checks whether the chunking results would exceed the max chunk limit.
-     * If exceeds, then Throw IllegalStateException
+     * Checks whether the chunking results would exceed the max chunk limit after adding a passage
+     * If exceeds, then return true
      *
      * @param chunkResultSize the size of chunking result
      * @param runtimeMaxChunkLimit runtime max_chunk_limit, used to check with chunkResultSize
-     * @param nonRuntimeMaxChunkLimit non-runtime max_chunk_limit, used to keep exception message consistent
+     * @param stringTobeChunkedCount runtime string_tobe_chunked_count, used to check with chunkResultSize
      */
-    public static void checkRunTimeMaxChunkLimit(int chunkResultSize, int runtimeMaxChunkLimit, int nonRuntimeMaxChunkLimit) {
-        if (runtimeMaxChunkLimit != DISABLED_MAX_CHUNK_LIMIT && chunkResultSize >= runtimeMaxChunkLimit) {
-            throw new IllegalArgumentException(
-                String.format(
-                    Locale.ROOT,
-                    "The number of chunks produced by %s processor has exceeded the allowed maximum of [%s]. This limit can be set by changing the [%s] parameter.",
-                    TYPE,
-                    nonRuntimeMaxChunkLimit,
-                    MAX_CHUNK_LIMIT_FIELD
-                )
-            );
-        }
+    public static boolean checkRunTimeMaxChunkLimit(int chunkResultSize, int runtimeMaxChunkLimit, int stringTobeChunkedCount) {
```

Comment:
Can we move this method to default method in interface? Creating a new class file for one method seems overheading.

Reply:
Good point. I will update it.